pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -476,6 +478,23 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 	for _, osd := range osds {
 		logger.Debugf("start osd %v", osd)
 
+		// create encryption Kubernetes Secret if the osd is encrypted
+		if osd.EncryptionKey != "" {
+			// Generate the encrypted key Kubernetes Secret
+			s := generateOSDEncryptedKeySecret(osd.ID, c.clusterInfo.Namespace, osd.EncryptionKey)
```

Comment:
Doesn't the encryption key have to be generated before the osd prepare job? This looks like it's with the osd daemon, after the prepare.

Reply:
The flow is:

* The key is generated in the `provisionOSDContainer()`
* then passed as an env var in the provision pod
* c-v uses it
* we stored the key in `osdInfo` (CM) unfortunately not so secure, the initial implementation was more secure from that point of view
* before starting the osd we store the key in a secret named after the OSD ID
* start the OSD

Reply:
Solved as discussed using the pvc name.